File: src/004. Largest palindrome product/004.rb
class Integer
  def is_palindrome?
    return self == self.to_s.reverse.to_i
  end
end

max_n = 0
999.downto(100) do |a|
  999.downto(100) do |b|
    n = a * b
    if n > max_n and n.is_palindrome?
      max_n = n
    end
  end
end

puts max_n

File: src/004. Largest palindrome product/004.py
def is_palindrome(n):
    return str(n)[::-1] == str(n)

max_n = 0
for a in range(999, 100, -1):
    for b in range(999, 100, -1):
        n = a * b
        if n > max_n and is_palindrome(n):
            max_n = n
print(max_n)
